maestro-cli/src/main/java/maestro/cli/runner/MaestroCommandRunner.kt

```diff
@@ -43,25 +47,55 @@ object MaestroCommandRunner {
         view: ResultView,
         commands: List<MaestroCommand>,
         cachedAppState: OrchestraAppState?,
+        debug: FlowDebugMetadata
     ): Result {
         val initFlow = YamlCommandReader.getConfig(commands)?.initFlow
 
         val commandStatuses = IdentityHashMap<MaestroCommand, CommandStatus>()
         val commandMetadata = IdentityHashMap<MaestroCommand, Orchestra.CommandMetadata>()
 
+        // debug
+        val debugCommands = debug.commands
+        val debugScreenshots = debug.screenshots
+
+        fun takeDebugScreenshot(status: CommandStatus): File? {
+            val containsFailed = debugScreenshots.any { it.status == CommandStatus.FAILED }
```

Comment:
Is this function intended to be called when flow fails?

Reply:
Yes. Initially, it was for more steps but it created too many screenshots. For now it's for failure only.